test/powershell/Language/Operators/ReplaceOperator.Tests.ps1

```diff
@@ -0,0 +1,67 @@
+Describe "Replace Operator" -Tags CI {
+    Context "Replace operator" {
+        It "Replace operator can replace string values using regular expressions" {
+            $res = "Get-Process" -replace "Get", "Stop"
+            $res | Should Be "Stop-Process"
+
+            $res = "image.gif" -replace "\.gif$",".jpg"
+            $res | Should Be "image.jpg"
+        }
+
+        It "Replace operator can be case-insensitive and case-sensitive" {
+            $res = "book" -replace "B","C"
+            $res | Should BeExactly "Cook"
+
+            $res = "book" -ireplace "B","C"
+            $res | Should BeExactly "Cook"
+
+            $res = "book" -creplace "B","C"
+            $res | Should BeExactly "book"
+        }
+
+        It "Replace operator can take 2 arguments, a mandatory pattern, and an optional substitution" {
+            $res = "PowerPoint" -replace "Point","Shell"
+            $res | Should Be "PowerShell"
+
+            $res = "PowerPoint" -replace "Point"
+            $res | Should Be "Power"
+        }
+    }
+
+    Context "Replace operator substitutions" {
+        It "Replace operator supports numbered substitution groups using ```$n" {
+            $res = "domain.example" -replace ".*\.(\w+)$","`$0"
+            $res | Should Be "domain.example"
+
+            $res = "domain.example" -replace ".*\.(\w+)$","`$1"
+            $res | Should Be "example"
```

Comment:
We could use one Should with `$res = "domain.example" -replace ".*\.(\w+)$","$0 - $1" `

Reply:
Done